### DISCUSSION TITLE
[功能建议] 视频DASH格式纯音频下载

### DISCUSSION THREAD

**Discussion by CKylinMC** [the1812/Bilibili-Evolved#2325] [功能建议 / Ideas]
DASH格式应该是音画分离下载后合并mp4，那能否在DASH下提供一个直接功能仅下载视频的音频版本？

~~实际上是我室友刚刚在问我有没有办法视频直接下载音频，因为他觉得每次下载视频再提取音频很繁琐也很费流量...~~

**Comment by the1812** [comment]
问过很多次的建议了...  v2 里的架构可以做到, 不过还没写对应的插件, ~~说不定你就能写一个~~

**Reply by CKylinMC** [reply]
~~我实际上fork了正在研究~~
 我想知道Evolved自己提供了哪些接口...

**Reply by the1812** [reply]
要不直接改内置的吧, 感觉另外写插件又要重复 DASH 的部分逻辑了
`registry/lib/components/video/download/apis/dash.ts` 里是 DASH 的下载逻辑, 你可以直接魔改下弄出仅音频的 `DownloadVideoApi`

**Reply by CKylinMC** [reply]
好家伙 我看看

**Reply by CKylinMC** [reply]
我提一个情况，刚刚我帮室友导出设置的时候，动态页面的链接污染把脚本导出设置的按钮也污染了...

**Reply by the1812** [reply]
那个会污染所有 `<a>` 的, 影响脚本所有的文件下载, 那个 issue 里有发链接, 直接 ABP 屏蔽掉就行了

**Comment by the1812** [comment]
看了下你 Fork 的仓库, 好像跟我想的不太一样... 增加音频的功能应该只需要在 `dash.ts` 中完成, 主模块 `DownloadVideo.vue` 把 `DownloadVideoApi` 导进去就可以了, 不应该变动其他模块
我等会传个实现音频的代码上去给你看看吧

**Reply by CKylinMC** [reply]
啊这 我还没仔细看...  先谢谢了

**Reply by the1812** [reply]
https://github.com/the1812/Bilibili-Evolved/commit/119bcb5711b67d7c32c2bef4ad91e1b7b31631b4
